libkbfs/block_ops.go

```diff
@@ -14,10 +16,23 @@ import (
 // requests to the block server.
 type BlockOpsStandard struct {
 	config Config
+	bwKBps int
+
+	lock sync.Mutex
 }
 
 var _ BlockOps = (*BlockOpsStandard)(nil)
 
+// NewBlockOpsStandard constructs a new standard implementation of the
+// BlockOps interface. bwKBps indicates a constrained bandwidth to
+// simulate in kilobytes per second (0 means unconstrained).
+func NewBlockOpsStandard(config Config, bwKBps int) *BlockOpsStandard {
```

Comment:
Can you instead implement that bandwidth constraint as a separate type that embeds a `BlockOps`? Something like:

```
type BlockOpsConstrained struct {
  BlockOps
  lock sync.Mutex
  bwKBps int
}

func ... Put(...) {
  b.Lock()
  time.Sleep(...)
  b.Unlock()
  return b.BlockOps.Put(...)
}
```